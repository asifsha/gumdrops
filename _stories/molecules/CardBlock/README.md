The `<CardBlock>` component should be nested within a `<Card>` component to wrap the card's content into distinct chunks.

Use the `option` prop to add a divider line on the top or bottom of the `<CardBlock>` component, to make the separation more clear.
