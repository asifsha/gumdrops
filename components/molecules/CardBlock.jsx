import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const CardBlock = ({ option, className, style, children }) => {
    const baseClass = 'gds-card__block';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option
    });

    return (
        <div className={rootClass} style={style}>
            {children}
        </div>
    );
};

CardBlock.displayName = 'CardBlock';

CardBlock.propTypes = {
    /** divide-top, divide-bottom */
    option: PropTypes.oneOf(['divide-top', 'divide-bottom']),
    className: PropTypes.string,
    style: PropTypes.object,
    children: PropTypes.node
};

export default CardBlock;
