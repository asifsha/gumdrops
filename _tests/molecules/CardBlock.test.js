/* globals mount */
import React from 'react';
import CardBlock from '../../components/molecules/CardBlock';

const defaultProps = {
    option: 'divide-top',
    className: 'custom',
    style: { width: 120 }
};

describe('Expect <CardBlock>', () => {
    it('to render', () => {
        const wrapper = mount(
            <CardBlock {...defaultProps}>
                <div>Foo</div>
            </CardBlock>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
