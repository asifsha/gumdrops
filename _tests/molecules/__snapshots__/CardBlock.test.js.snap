// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <CardBlock> to render 1`] = `
<CardBlock
  className="custom"
  option="divide-top"
  style={
    Object {
      "width": 120,
    }
  }
>
  <div
    className="gds-card__block custom gds-card__block--divide-top"
    style={
      Object {
        "width": 120,
      }
    }
  >
    <div>
      Foo
    </div>
  </div>
</CardBlock>
`;
